src/instruments/src/EFB/Ground/Pages/ServicesPage.tsx

```diff
@@ -38,159 +43,524 @@ const ServiceButtonWrapper: FC<ServiceButtonWrapperProps> = ({ children, classNa
 
     // Wheel Chocks and Cones
     const [isGroundEquipmentVisible] = useSimVar('L:A32NX_GND_EQP_IS_VISIBLE', 'bool', 500);
     const [wheelChocksEnabled] = useSimVar('L:A32NX_MODEL_WHEELCHOCKS_ENABLED', 'bool', 500);
     const [conesEnabled] = useSimVar('L:A32NX_MODEL_CONES_ENABLED', 'bool', 500);
-    const [wheelChocksVisible, setWheelChocksVisible] = useState(wheelChocksEnabled && isGroundEquipmentVisible);
-    const [conesVisible, setConesVisible] = useState(conesEnabled && isGroundEquipmentVisible);
+    const wheelChocksVisible = wheelChocksEnabled && isGroundEquipmentVisible;
+    const conesVisible = conesEnabled && isGroundEquipmentVisible;
 
-    const [tugActive] = useState(false);
-
-    const STATE_WAITING = 'WAITING';
+    // Service events
+    const toggleCabinDoor = () => SimVar.SetSimVarValue('K:TOGGLE_AIRCRAFT_EXIT', 'enum', 1);
+    const toggleJetBridgeAndStairs = () => {
+        SimVar.SetSimVarValue('K:TOGGLE_JETWAY', 'bool', false);
+        SimVar.SetSimVarValue('K:TOGGLE_RAMPTRUCK', 'bool', false);
+    };
+    const toggleCargoDoor = () => SimVar.SetSimVarValue('K:TOGGLE_AIRCRAFT_EXIT', 'enum', 6);
+    const toggleBaggageTruck = () => SimVar.SetSimVarValue('K:REQUEST_LUGGAGE', 'bool', true);
+    const toggleAftDoor = () => SimVar.SetSimVarValue('K:TOGGLE_AIRCRAFT_EXIT', 'enum', 4);
+    const toggleCateringTruck = () => SimVar.SetSimVarValue('K:REQUEST_CATERING', 'bool', true);
+    const toggleFuelTruck = () => SimVar.SetSimVarValue('K:REQUEST_FUEL_KEY', 'bool', true);
+    const toggleGpu = () => SimVar.SetSimVarValue('K:REQUEST_POWER_SUPPLY', 'bool', true);
 
-    useEffect(() => {
-        setGroundServicesAvailable(simOnGround && aircraftIsStationary);
-    }, [simOnGround, aircraftIsStationary]);
+    // Button states
+    const cabinDoorButtonState = useAppSelector((state) => state.groundServicePage.cabinDoorButtonState);
+    const jetWayButtonState = useAppSelector((state) => state.groundServicePage.jetWayButtonState);
+    const fuelTruckButtonState = useAppSelector((state) => state.groundServicePage.fuelTruckButtonState);
+    const gpuButtonState = useAppSelector((state) => state.groundServicePage.gpuButtonState);
+    const cargoDoorButtonState = useAppSelector((state) => state.groundServicePage.cargoDoorButtonState);
+    const baggageButtonState = useAppSelector((state) => state.groundServicePage.baggageButtonState);
+    const aftDoorButtonState = useAppSelector((state) => state.groundServicePage.aftDoorButtonState);
+    const cateringButtonState = useAppSelector((state) => state.groundServicePage.cateringButtonState);
 
-    useEffect(() => {
-        setWheelChocksVisible(wheelChocksEnabled && isGroundEquipmentVisible);
-        setConesVisible(conesEnabled && isGroundEquipmentVisible);
-    }, [isGroundEquipmentVisible, wheelChocksEnabled, conesEnabled]);
+    // Required so these can be used inside the useTimeout callback
```

Comment:
Can you explain why you have had to wrap these states in refs? What issues in particular have you encountered in trying to use the states directly?

Reply:
They simply had the old state when the timeout was called - tried all the methods I knew - added a parameter array, adding parameter, etc. - the useTimeout used the variable state when defining the timeout instead the state of the variables when executing it. The refs where the easiest and most reliable way to get the current value.

Reply:
@ErickSharp you want to look more into this or is this ok as it is?

Reply:
@ErickSharp bump 😀

Reply:
@frankkopp im very sorry! I haven’t been at my computer for a while. Yes, this is fine with me 🙂